cesm/nuopc_cap_share/driver_pio_mod.F90

```diff
@@ -212,86 +212,97 @@ subroutine driver_pio_component_init(driver, ncomps, rc)
        if (ESMF_GridCompIsPetLocal(gcomp(i), rc=rc)) then
           call ESMF_GridCompGet(gcomp(i), vm=vm, name=cval, rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
+          
           io_compname(i) = trim(cval)
-
+          
           call NUOPC_CompAttributeAdd(gcomp(i), attrList=(/'MCTID'/), rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
-
+          
           write(cval, *) io_compid(i)
           call NUOPC_CompAttributeSet(gcomp(i), name="MCTID", value=trim(cval), rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-          call ESMF_VMGet(vm, mpiCommunicator=comp_comm, localPet=comp_rank, petCount=npets, &
-               ssiLocalPetCount=default_stride, rc=rc)
+          call ESMF_VMGet(vm, mpiCommunicator=comp_comm, rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
-          
-          call NUOPC_CompAttributeGet(gcomp(i), name="pio_stride", value=cval, rc=rc)
-          if (chkerr(rc,__LINE__,u_FILE_u)) return
-          read(cval, *) pio_comp_settings(i)%pio_stride
-          if(pio_comp_settings(i)%pio_stride <= 0 .or. pio_comp_settings(i)%pio_stride > npets) then
-             pio_comp_settings(i)%pio_stride = min(npets, default_stride)
-          endif
-          
-          call NUOPC_CompAttributeGet(gcomp(i), name="pio_rearranger", value=cval, rc=rc)
-          if (chkerr(rc,__LINE__,u_FILE_u)) return
-          read(cval, *) pio_comp_settings(i)%pio_rearranger
-          
-          call NUOPC_CompAttributeGet(gcomp(i), name="pio_numiotasks", value=cval, rc=rc)
-          if (chkerr(rc,__LINE__,u_FILE_u)) return
-          read(cval, *) pio_comp_settings(i)%pio_numiotasks
-          
-          if(pio_comp_settings(i)%pio_numiotasks < 0 .or. pio_comp_settings(i)%pio_numiotasks > npets) then
-             pio_comp_settings(i)%pio_numiotasks = max(1,npets/pio_comp_settings(i)%pio_stride)
-          endif
 
+          if(comp_comm .ne. MPI_COMM_NULL) then
+             call ESMF_VMGet(vm, petCount=npets, rc=rc)
+             if (chkerr(rc,__LINE__,u_FILE_u)) return
 
-          call NUOPC_CompAttributeGet(gcomp(i), name="pio_root", value=cval, rc=rc)
-          if (chkerr(rc,__LINE__,u_FILE_u)) return
-          read(cval, *) pio_comp_settings(i)%pio_root
-          
-          if(pio_comp_settings(i)%pio_root < 0 .or. pio_comp_settings(i)%pio_root > npets) then
-             pio_comp_settings(i)%pio_root = 0
-          endif
-          
+             call ESMF_VMGet(vm, localPet=comp_rank, rc=rc) 
+             if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+             call ESMF_VMGet(vm, ssiLocalPetCount=default_stride, rc=rc)
+             if (chkerr(rc,__LINE__,u_FILE_u)) return
```

Comment:
Is there a reason why you use multiple calls here instead of a single call with multiple arguments?

Reply:
thanks for pointing that out.